lint/nestingReduce_checker.go

```diff
@@ -0,0 +1,70 @@
+package lint
+
+import "go/ast"
+
+//! Finds where nesting level could be reduced.
+//
+// @Before:
+// for _, v := range a {
+//    if v.Bool {
+//        ...
+//    }
+// }
+//
+// @After:
+// for _, v := range a {
+//    if ! v.Bool {
+//        continue
+//    }
+//    ...
+// }
+
+func init() {
+	addChecker(&nestingReduceChecker{}, attrExperimental)
+}
+
+type nestingReduceChecker struct {
+	checkerBase
+}
+
+func (c *nestingReduceChecker) VisitFuncDecl(decl *ast.FuncDecl) {
+	if decl.Body == nil {
+		return
+	}
+	c.checkBody(decl.Body.List)
+	for _, stmt := range decl.Body.List {
+		switch stmt := stmt.(type) {
+		case *ast.RangeStmt:
+			c.checkBody(stmt.Body.List)
+		case *ast.ForStmt:
+			c.checkBody(stmt.Body.List)
+		case *ast.IfStmt:
+			c.checkBody(stmt.Body.List)
+		default:
+		}
+	}
+}
+
+func (c *nestingReduceChecker) checkBody(body []ast.Stmt) {
+	if len(body) == 1 {
+		if stmt, ok := body[0].(*ast.IfStmt); ok && c.checkIf(stmt) {
+			c.warn(stmt)
+		}
+		return
+	}
```

Comment:
Almost sure nestingReduce checker should detect this function. :)
(But I guess it's temporary/initial implementation, so it's not how it will look like eventually.)

Reply:
Right now it wouldn't :)